src/memory/AmdFlash.cc

```diff
@@ -52,6 +53,9 @@ AmdFlash::AmdFlash(const std::string& name, std::span<const SectorInfo> sectorIn
 void AmdFlash::init(const std::string& name, const DeviceConfig& config, Load load, const Rom* rom)
 {
 	assert(std::has_single_bit(size()));
+	assert(std::popcount(to_underlying(chip.autoSelect.manufacturer)) & 1); // check parity
+	assert(chip.autoSelect.device[0] == 0x7E ||
+		(chip.autoSelect.device[1] == 0x00 && chip.autoSelect.device[2] == 0x00)); // single or triple
```

Comment:
I'll try to review commit per commit, but sometimes also looking ahead to the final code. E.g. in this case I saw that several more asserts on `chip` were added.

Detail:
Would it be possible (and easy) to move these kind of checks to compile-time? Then these things are caught at compile-time instead of only at-run-time and then only when the chip is actually being used? Maybe add a `constexpr Chip` constructor??

Reply:
If I add a constructor I can’t use designated initializers anymore. I think I’ll wrap it in a `constexpr Chip validate(Chip&& chip)` function or something like that.